modules/globebrowsing/tile/rawtiledatareader/simplerawtiledatareader.cpp

```diff
@@ -0,0 +1,186 @@
+
+namespace {
+    const std::string _loggerCat = "SimpleRawTileDataReader";
+}
+
+namespace openspace {
+namespace globebrowsing {
+
+SimpleRawTileDataReader::SimpleRawTileDataReader(
+    const std::string& filePath, const Configuration& config)
+    : RawTileDataReader(config)
+{
+    _datasetFilePath = filePath;
+    ensureInitialized();
+}
+
+SimpleRawTileDataReader::~SimpleRawTileDataReader() {
+    
+}
+
+void SimpleRawTileDataReader::reset() {
+    initialize();
+}
+
+int SimpleRawTileDataReader::maxChunkLevel() {
+    return _cached._maxLevel;
+}
+
+float SimpleRawTileDataReader::noDataValueAsFloat() const {
+    return 0.0f;
+}
+
+int SimpleRawTileDataReader::rasterXSize() const {
+    return _dataTexture->dimensions().x;
+}
+
+int SimpleRawTileDataReader::rasterYSize() const {
+    return _dataTexture->dimensions().y;
+}
+
+float SimpleRawTileDataReader::depthOffset() const {
+    return 0.0f;
+}
+
+float SimpleRawTileDataReader::depthScale() const {
+    return 1.0f;
+}
+
+IODescription SimpleRawTileDataReader::getIODescription(const TileIndex& tileIndex) const {
+
+    IODescription io;
+    io.read.overview = 0;
+    io.read.region = highestResPixelRegion(tileIndex);
+    io.read.fullRegion = PixelRegion({0, 0}, {rasterXSize(), rasterYSize()});
+    io.write.region = PixelRegion({0, 0}, io.read.region.numPixels);
+    io.write.bytesPerLine = _dataTexture->bytesPerPixel() * io.write.region.numPixels.x;
+    io.write.totalNumBytes = io.write.bytesPerLine * io.write.region.numPixels.y;
+    
+    return io;
+}
+
+void SimpleRawTileDataReader::initialize() {
+    _dataTexture = ghoul::io::TextureReader::ref().loadTexture(_datasetFilePath);
+    if (_dataTexture == nullptr) {
+        throw ghoul::RuntimeError(
+            "Unable to read dataset: " + _datasetFilePath +
+            ".\nCompiling OpenSpace with GDAL will allow for better support for different formats.");
+    }
+    float exponentX = log2(_dataTexture->dimensions().x);
+    float exponentY = log2(_dataTexture->dimensions().y);
+    if ( (exponentX - static_cast<int>(exponentX)) > 0.0001 ||
+       (exponentY - static_cast<int>(exponentY)) > 0.0001 ) {
+        throw ghoul::RuntimeError(
+            "Unable to read dataset: " + _datasetFilePath +
+            ".\nCurrently only supporting power of 2 textures.");
+    }
+  
+    _cached._maxLevel = 2;
+    _cached._tileLevelDifference = 0;
+
+    _dataLayout.glType = _dataTexture->dataType();
+    _dataLayout.bytesPerDatum = tiledatatype::numberOfBytes(_dataLayout.glType);
+    _dataLayout.numRasters = tiledatatype::numberOfRasters(_dataTexture->format());
+    _dataLayout.numRastersAvailable = _dataLayout.numRasters;
+    _dataLayout.bytesPerPixel = _dataLayout.bytesPerDatum * _dataLayout.numRasters;
+    _dataLayout.textureFormat = {_dataTexture->format(), _dataTexture->internalFormat()};
+
+    _depthTransform = {depthScale(), depthOffset()};
+}
+
+char* SimpleRawTileDataReader::readImageData(
```

Comment:
The way it is right now the pointer is propagated up to `ghoul::opengl::Texture` which takes ownership of the data. The Texture class only takes a raw pointer for the data. Wouldn't it be weird to have a vector of chars somewhere else which the Texture uses and don't have ownership of?